istioctl/pkg/ztunnelconfig/ztunnelconfig.go

```diff
@@ -0,0 +1,486 @@
+	ztunnelDump "istio.io/istio/istioctl/pkg/writer/ztunnel/configdump"
+	"istio.io/istio/pkg/kube"
+	"istio.io/istio/pkg/log"
+)
+
+const (
+	jsonOutput    = "json"
+	yamlOutput    = "yaml"
+	summaryOutput = "short"
+
+	defaultProxyAdminPort = 15000
+)
+
+var (
+	workloadsNamespace string
+	verboseProxyConfig bool
+
+	address, node string
+
+	// output format (json, yaml or short)
+	outputFormat string
+
+	proxyAdminPort int
+
+	configDumpFile string
+
+	labelSelector = ""
+)
+
+func ZtunnelConfig(ctx cli.Context) *cobra.Command {
+	configCmd := &cobra.Command{
+		Use:   "ztunnel-config",
+		Short: "Retrieve information about ztunnel configuration.",
+		Long:  "A group of commands used to retrieve information about Ztunnel configuration from a Ztunnel instance.",
+		Example: `  # Retrieve summary about workload configuration for a given ztunnel.
+	istioctl ztunnel-config <workload> <ztunnel-name[.namespace]>`,
+		Aliases: []string{"zc"},
+	}
+
+	configCmd.PersistentFlags().StringVarP(&outputFormat, "output", "o", summaryOutput, "Output format: one of json|yaml|short")
+	configCmd.PersistentFlags().IntVar(&proxyAdminPort, "proxy-admin-port", defaultProxyAdminPort, "Envoy proxy admin port")
+
+	configCmd.AddCommand(logCmd(ctx))
+	configCmd.AddCommand(workloadConfigCmd(ctx))
+
+	return configCmd
+}
+
+func workloadConfigCmd(ctx cli.Context) *cobra.Command {
+	var podName, podNamespace string
+
+	workloadConfigCmd := &cobra.Command{
+		Use:   "workload [<type>/]<name>[.<namespace>]",
+		Short: "Retrieves workload configuration for the specified ztunnel pod",
+		Long:  `Retrieve information about workload configuration for the ztunnel instance.`,
+		Example: `  # Retrieve summary about workload configuration for a given ztunnel.
+  istioctl proxy-config workload <ztunnel-name[.namespace]>
+
+  # Retrieve summary of workloads on node XXXX for a given ztunnel instance.
+  istioctl proxy-config workload <ztunnel-name[.namespace]> --node ambient-worker
+
+  # Retrieve full workload dump of workloads with address XXXX for a given ztunnel
+  istioctl proxy-config workload <ztunnel-name[.namespace]> --address 0.0.0.0 -o json
+
+  # Retrieve workload summary
+  kubectl exec -it $ZTUNNEL -n istio-system -- curl localhost:15000/config_dump > ztunnel-config.json
+  istioctl proxy-config workloads --file ztunnel-config.json
+
+  # Retrieve workload summary for a specific namespace
+  istioctl proxy-config workloads <ztunnel-name[.namespace]> --workloads-namespace foo
+`,
+		Aliases: []string{"workloads", "w"},
+		Args: func(cmd *cobra.Command, args []string) error {
+			if (len(args) == 1) != (configDumpFile == "") {
+				cmd.Println(cmd.UsageString())
+				return fmt.Errorf("workload requires pod name or --file parameter")
+			}
+			return nil
+		},
+		RunE: func(c *cobra.Command, args []string) error {
+			kubeClient, err := ctx.CLIClient()
+			if err != nil {
+				return err
+			}
+			var configWriter *ztunnelDump.ConfigWriter
+			if len(args) == 1 {
```

Comment:
can be a followup: We should have the same pod identification logic in `workload` and `log`. Currently it differs. We will also probably have future commands so some abstraction here will be nice

Reply:
Agreed! I can followup with this.